benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/tasks/start/StartTask.java

```diff
@@ -31,49 +26,55 @@
   private static Logger logger = LoggerFactory.getLogger(StartTask.class.getSimpleName());
 
   private final String testID;
-  private final String testDefinitionYamlString;
-  private final InputStream deploymentDescriptorInputStream;
-  private final Map<String, InputStream> bpmnModelInputStreams;
 
   // services
   private final MinioService minioService;
   private final ExplorationModelDAO explorationModelDAO;
-  private final BenchFlowTestTaskController testTaskController;
 
-  public StartTask(String testID, String testDefinitionYamlString,
-      InputStream deploymentDescriptorInputStream, Map<String, InputStream> bpmnModelInputStreams) {
+  public StartTask(String testID) {
 
     this.testID = testID;
-    this.testDefinitionYamlString = testDefinitionYamlString;
-    this.deploymentDescriptorInputStream = deploymentDescriptorInputStream;
-    this.bpmnModelInputStreams = bpmnModelInputStreams;
 
     this.minioService = BenchFlowTestManagerApplication.getMinioService();
     this.explorationModelDAO = BenchFlowTestManagerApplication.getExplorationModelDAO();
-    this.testTaskController = BenchFlowTestManagerApplication.getTestTaskController();
+
+  }
+
+  public static List<Integer> generateExplorationSpace(BenchFlowTest test) {
+
+    // generate exploration space if any
+
+    // TODO - replace this with calculating all possible combinations
+    // something like this https://blog.balfes.net/2015/06/08/finding-every-possible-combination-of-array-entries-from-multiple-lists-with-unknown-bounds-in-java/
+
+    if (test.configuration().goal().explorationSpace().isDefined()) {
+
+      if (test.configuration().goal().explorationSpace().get().workload().isDefined()) {
```

Comment:
The workload part of the exploration space is not always defined, e.g. when we just want to change resources. It will move to the DSL in a new pull-request.